diff --color=auto -Naur a/modules/system/defaults/dock.nix b/modules/system/defaults/dock.nix
--- a/modules/system/defaults/dock.nix	2024-09-04 19:37:51
+++ b/modules/system/defaults/dock.nix	2024-09-05 21:39:12
@@ -126,27 +126,62 @@
     system.defaults.dock.persistent-apps = mkOption {
       type = types.nullOr (types.listOf (types.either types.path types.str));
       default = null;
-      example = [ "/Applications/Safari.app" "/System/Applications/Utilities/Terminal.app" ];
+      example = [
+        "/Applications/Safari.app"
+        "/System/Applications/Utilities/Terminal.app"
+      ];
       description = ''
         Persistent applications in the dock.
       '';
       apply = value:
-        if !(isList value)
-        then value
-        else map (app: { tile-data = { file-data = { _CFURLString = app; _CFURLStringType = 0; }; }; }) value;
+        if !(isList value) then
+          value
+        else
+          map (app: {
+            tile-data = {
+              file-data = {
+                _CFURLString = app;
+                _CFURLStringType = 0;
+              };
+            };
+          }) value;
     };
 
     system.defaults.dock.persistent-others = mkOption {
-      type = types.nullOr (types.listOf (types.either types.path types.str));
+      type = types.nullOr (types.listOf
+        (types.either types.attrs (types.oneOf [ types.path types.str types.attrs])));
       default = null;
-      example = [ "~/Documents" "~/Downloads" ];
+      example = [ "~/Documents" "~/Downloads" { "name" = "~/Music"; "displayas" = 1; } ];
       description = ''
         Persistent folders in the dock.
       '';
       apply = value:
-        if !(isList value)
-        then value
-        else map (folder: { tile-data = { file-data = { _CFURLString = "file://" + folder; _CFURLStringType = 15; }; }; tile-type = if strings.hasInfix "." (last (splitString "/" folder)) then "file-tile" else "directory-tile"; }) value;
+        if !(isList value) then  # it's a null
+          value
+        else
+          map (arg:
+            let
+              target = if (isAttrs arg) then arg.name else arg;
+              displayas = if (isAttrs arg) then (arg.displayas or 0) else 0;
+              tile-type = if (isAttrs arg) then
+                arg."tile-type" or default-tile-type
+              else
+                default-tile-type;
+              default-tile-type =
+                if strings.hasInfix "." (last (splitString "/" target)) then
+                  "file-tile"
+                else
+                  "directory-tile";
+            in {
+              tile-data = {
+                displayas = displayas;
+                file-data = {
+                  _CFURLString = "file://" + target;
+                  _CFURLStringType = 15;
+                };
+              };
+              tile-type = tile-type;
+            }) value;
     };
 
     system.defaults.dock.show-process-indicators = mkOption {
@@ -204,7 +239,6 @@
         Magnified icon size on hover. The default is 16.
       '';
     };
-   
 
     system.defaults.dock.wvous-tl-corner = mkOption {
       type = types.nullOr types.ints.positive;
@@ -290,6 +324,6 @@
       '';
     };
 
-    };
+  };
 }
 
