--- dock.nix.orig	2024-08-23 16:38:34
+++ dock.nix	2024-08-23 16:41:48
@@ -146,7 +146,7 @@
       apply = value:
         if !(isList value)
         then value
-        else map (folder: { tile-data = { file-data = { _CFURLString = "file://" + folder; _CFURLStringType = 15; }; }; tile-type = if strings.hasInfix "." (last (splitString "/" folder)) then "file-tile" else "directory-tile"; }) value;
+        else map (folder: { tile-data = { file-data = { _CFURLString = "file://" + folder; _CFURLStringType = 15; }; display-as = 1; }; tile-type = if strings.hasInfix "." (last (splitString "/" folder)) then "file-tile" else "directory-tile"; }) value;
     };
 
     system.defaults.dock.show-process-indicators = mkOption {
